spec/features/visitors/email_language_preference_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+describe 'visitor signs up with email language preference' do
+  it 'defaults to the current locale' do
+    visit sign_up_email_path
+
+    field = page.find_field(
+      t('account.email_language.default', language: t("i18n.locale.#{I18n.default_locale}")),
+    )
+    expect(field).to be_present
+    expect(field[:lang]).to eq(I18n.default_locale.to_s)
+    (I18n.available_locales - [I18n.default_locale]).each do |locale|
+      field = page.find_field(t("i18n.locale.#{locale}"))
+      expect(field).to be_present
+      expect(field[:lang]).to eq(locale.to_s)
+    end
+
+    visit sign_up_email_path(:es)
+
+    field = page.find_field(t('account.email_language.default', language: t('i18n.locale.es')))
+    expect(field).to be_present
+    expect(field[:lang]).to eq('es')
+    (I18n.available_locales - [:es]).each do |locale|
+      field = page.find_field(t("i18n.locale.#{locale}"))
+      expect(field).to be_present
+      expect(field[:lang]).to eq(locale.to_s)
+    end
+  end
+
+  it 'sends emails in the selected language' do
```

Comment:
For the ticket, do we need a test for email sending?

Reply:
> For the ticket, do we need a test for email sending?

Not strictly in scope, no, but it didn't look like we had any existing coverage, so added while I was here, since I didn't have high hopes coverage would ever be added otherwise.